dbt_semantic_interfaces/transformations/rule_set.py

```diff
@@ -0,0 +1,31 @@
+import logging
+from abc import abstractmethod
+from typing import Protocol, Sequence
+
+from dbt_semantic_interfaces.transformations.transform_rule import (
+    SemanticManifestTransformRule,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class SemanticManifestTransformRuleSet(Protocol):
+    """Groups rules that should be run for a SemanticManifest."""
+
+    @property
+    @abstractmethod
+    def primary_rules(self) -> Sequence[SemanticManifestTransformRule]:
+        """TODO: Define what primary means."""
+        pass
+
+    @property
+    @abstractmethod
+    def secondary_rules(self) -> Sequence[SemanticManifestTransformRule]:
+        """TODO: Define what secondary means."""
+        pass
```

Comment:
Secondary rules run after all primary rules are complete. The execution of the rules can happen in arbitrary order within each level, but primary will always block secondary.

In practice I think we have primary "basic" transforms that make sense to apply before doing anything else.